src/nl/hannahsten/texifyidea/util/CommandRunner.kt

```diff
@@ -0,0 +1,127 @@
+package nl.hannahsten.texifyidea.util
+
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.util.io.awaitExit
+import kotlinx.coroutines.*
+import java.io.File
+import java.io.IOException
+import javax.swing.SwingUtilities
+
+data class CommandResult(
+    val exitCode: Int,
+    val standardOutput: String?,
+    val errorOutput: String?
+) {
+
+    val output: String?
+        get() = if (standardOutput != null || errorOutput != null) standardOutput + errorOutput else null
+}
+
+/**
+ * Run a command in the terminal in a non-blocking way.
+ *
+ * @param workingDirectory If provided, the process' working directory.
+ * @param input If provided, this will be written to the process' input pipe.
+ * @param discardOutput Whether to discard all command outputs (stdout, stderr) and only return its exit code.
+ * @param returnExceptionMessageAsErrorOutput Whether to return exception messages as error output if exceptions are thrown.
+ * @param timeout The timeout for execution. Does not stop reading the process' output as long as it is available.
+ */
+suspend fun runCommandNonBlocking(
```

Comment:
> but I don't know how nl.hannahsten.texifyidea.util.files.LatexPackageLocationCache#getPackageLocation() is called. 

If I remove LatexPackageLocationCacheInitializer then it doesn't appear to be blocking the UI, at least it's running in a background thread.

Reply:
Good idea on how to test this.

Sadly I was now able to provoke a freezed UI. Steps to reproduce:

1. Disable `LatexPackageLocationCacheInitializer`.
2. Add to document following content
```latex
\documentclass{report}

\begin{document}
\end{document}
```
3. Edit the file.

To make the effect more obvious, replace `runBlocking { fillCacheWithKpsewhich(project) }` with `runBlocking { delay(1_000) }`.

I can try to move the execution in another thread, but if this works depends on where the freezing occurs. Maybe another option is just to remove this call to `fillCacheWithKpsewhich()` as the cache should be filled at startup and `getPackageLocation()` is called multiple times - at least if it returns `null`. In either case I probably won't be able to evaluate this before next weekend.

Reply:
You're right, that does block the UI. But I don't understand, why is the UI not blocked if I put there `runBlocking { runCommandNonBlocking("sleep", "3") }` to simulate a long-running command? 

I'm also not quite sure why the UI is blocked if we run `runBlocking { delay(1_000) }` in a background thread. Documentation: https://plugins.jetbrains.com/docs/intellij/threading-model.html#avoiding-ui-freezes

It's true that for this particular case we can maybe circumvent the issue, but there are other places where we are running system calls and need to wait for the result to return it to IntelliJ api, and I don't know how to do that without blocking the thread (and thus the UI, apparently?). Another example is LatexDocumentationProvider, which can be triggered in the autocompletion popup. (If I put a delay in runCommandWithExitCode then UI will block)

In any case, that is out of scope for this PR, so I'll merge this already.

Reply:
Did you make sure the command is executed? I had to circumvent the if-clause to test this:
```kotlin
if (true || cache == null) {
    runBlocking {
        runCommandNonBlocking("ping", "-n", "30", "127.0.0.1", discardOutput = true)
    }
}
```

With this I was able to block the UI until timeout.

As to why it blocks the UI, I'm not really sure as the debugger implies that those are indeed run in separate threads.